packages/rolldown/src/utils/normalize-hook.ts

```diff
@@ -1,25 +1,20 @@
-import type { ObjectHook } from '../plugin'
-import type { AnyFn, AnyObj } from '../types/utils'
+import type { ObjectHook } from "../plugin";
 
-type NotFn<T> = T extends AnyFn ? never : T
+export function normalizeHook<T extends ObjectHook<Function>>(
+	hook: T,
+): T extends ObjectHook<infer A, infer B>
+	? T extends Function
+		? [A, {}]
+		: [A, Omit<B, "handler">]
+	: never {
+	if (typeof hook === "function") {
+    // @ts-ignore
+		return [hook, {}];
```

Comment:
previously: 
![image](https://github.com/user-attachments/assets/a7445366-5ea9-491d-a02a-765d042213f4)
current: 
![image](https://github.com/user-attachments/assets/01fdbd25-f147-4336-aa9e-e01cf651bab5)

Reply:
Sadly, it still needs ts **directive**, forgive me poor **type gymnastics**